browserdebuggertools/chrome/chromeExtensions/requestBlocker/requestBlocker.js

```diff
@@ -0,0 +1,27 @@
+function blockMainFrames() {
+    chrome.declarativeNetRequest.updateDynamicRules(
+        {
+            addRules: [{
+                "id": 1,
+                "priority": 1,
+                "action": {"type": "block"},
+                "condition": {"resourceTypes": ["main_frame"]}
+            }],
+            removeRuleIds: [1]
```

Comment:
It looks to me like we should be defining a static rule, and enable/disable it, instead of using dynamic rules. https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#method-updateEnabledRulesets